src/query.ts

```diff
@@ -204,231 +243,174 @@ export class Query {
+    this._page = (this._page || 1) + 1;
   }
 
   /**
    * Sets the Results Per Page (rpp) parameter.
    *
-   * @param  {Number}   rpp   The results per page to set
+   * @param  rpp - The results per page to set
    *
+   * @returns Query
    */
-  public resultsPerPage(rpp?: number): void {
-    if (rpp) {
-      this.setParameter('rpp', rpp);
+  public resultsPerPage(rpp?: number): Query {
+    if (typeof rpp === 'number' || typeof rpp === 'undefined') {
+      this._rpp = rpp;
     } else {
-      delete this.params.rpp;
+      throw new QueryValueError('Value error: Result per page must be a number');
     }
+    return this;
   }
 
   /**
    * Sets the types to query in this query, call without
    * parameters to clear the setting
    *
-   * @param  {String | String[]}  type    The type or types to set
-   *                                      for this query
-   *
-   */
-  public setTypes(type?: string | string[]): void {
-    if (type) {
-      this.setParameter('type', type);
-    } else {
-      delete this.params.type;
-    }
-  }
-
-  /**
-   * Adds a type to the current types in this query
-   *
-   * @param  {String}   type    The type to be added for the search
+   * @param  type - The type or types to set for this query
    *
+   * @returns Query
    */
-  public addType(type: string): void {
-    let typeParam: string[] = [];
-
-    if (typeof this.params.type === 'string') {
-      typeParam.push(this.params.type);
-    } else if (this.params.type) {
-      typeParam = this.params.type;
-    }
-
-    typeParam.push(type);
-    this.setParameter('type', typeParam);
-  }
-
-  /**
-   * Remove a type from this query
-   *
-   * @param  {String}   type    The type to be removed for the search
-   *
-   */
-  public removeType(type: string): void {
-    if (this.params.type) {
-      if (isArray(this.params.type)) {
-        const index: number = this.params.type.indexOf(type);
-        if (index !== -1) {
-          (this.params.type as Array<unknown>).splice(1, index);
+  public addTypes(types?: string | string[]): Query {
+    if (types) {
+      this._dataTypes = new Set();
+      if (typeof types === 'string') {
+        this._dataTypes.add(types);
+      } else if (isArray(types)) {
+        for (const type of types) {
+          this._dataTypes.add(type);
         }
-      } else if (this.params.type === type) {
-        delete this.params.type;
+      } else {
+        throw new QueryValueError('Value error: Types value must be string or array of strings');
       }
+    } else {
+      delete this._dataTypes;
     }
+    return this;
   }
 
   /**
    * Sets the transformer, call it empty to reset it to null
    *
-   * @param  {String}   transformer   The transformer option to set
+   * @param  transformer - The transformer option to set
+   *
+   * @returns Query
    */
-  public transformer(transformer?: TransformerOptions): void {
+  public transformer(transformer?: string): Query {
```

Comment:
This doesn't handle the case `query.transformer(null)`. And don't validate the parameter, just set it. This is a type of parameter we can customize for a user in the config and a validation won't let the user set it.